Telerik.JustMock/Expectations/NonPublicExpectationBase.cs

```diff
@@ -0,0 +1,238 @@
+/*
+ JustMock Lite
+ Copyright © 2010-2018 Telerik EAD
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using Telerik.JustMock.Core;
+using Telerik.JustMock.Core.Behaviors;
+using Telerik.JustMock.Core.Context;
+using Telerik.JustMock.Expectations.Abstraction;
+using Telerik.JustMock.Expectations.DynaMock;
+
+namespace Telerik.JustMock.Expectations
+{
+    internal abstract class NonPublicExpectationBase
```

Comment:
This class contains mostly methods that are used as helpers from other parts of the code, like NonPublicExpectation. Based on the functionality in the class I think that it is not very correct to  call this "NonPublicExpectationBase". Also I think that inheriting this class from NonPublicExpectation is not following the "IS A" OOP design rule. 

For example I have also moved this functionality outside NonPublicExpectation when implementing mocking for C# 7 Local Methods. In my case I named the class like "MockingUtil" and used it without inheritance. I think this approach is better in terms of OOP design.

Looking forward for your comments!

Reply:
@lyubomir-rusev, fixed with [595ded1](https://github.com/telerik/JustMockLite/pull/42/commits/595ded10a032d8e33576643110280741ea96a82d)